conn.go

```diff
@@ -257,12 +252,15 @@ func (c *Conn) Receive() ([]Message, error) {
 
 // receive is the internal implementation of Conn.Receive, which can be called
 // recursively to handle multi-part messages.
-func (c *Conn) receive() ([]Message, error) {
+func (c *Conn) receive() ([]Message, *OpError) {
```

Comment:
Returning the concrete `*OpError` seems dangerous in view of the possible "nil but not nil interface" mistake.

To expedite the conversation, I'm quoting your response from Slack:

> You raise a valid point; my thought here was to ensure that anything that comes out of this function has to be an OpError for consistency. I could potentially do something at the call site to check, or I could change this back and leave a note that all errors must be of type OpError or nil.

I like the latter option, but no strong opinions either way.